include/NAS2D/Filesystem.h

```diff
@@ -50,16 +57,64 @@ class Filesystem
-	Filesystem& operator= (const Filesystem&) = delete;
 
-	bool closeFile(void *file) const;
-	const char* getLastPhysfsError() const;
+    std::filesystem::path getExePath() const;
 
-private:
-	std::string			mDataPath;			/**< Data path string. This will typically be 'data/'. */
-	mutable bool		mVerbose;			/**< Displays lots of messages when true. Otherwise only critical messages are displayed. */
+    bool isInit() const;
+    bool exists(const std::filesystem::path& p) const;
+
+    void forEachFileInFolder(const std::filesystem::path& folderpath, const std::string& validExtensionList = std::string{}, const std::function<void(const std::filesystem::path&)>& callback = [](const std::filesystem::path& p) { (void)p; }, bool recursive = false) const;
+    void forEachFileInFolder(const std::filesystem::path& folderpath, const std::string& validExtensionList = std::string{}, const std::function<void(const std::filesystem::path&)>& callback = [](const std::filesystem::path& p) { (void)p; }, bool recursive = false);
+
+    template<typename DirectoryIteratorType>
+    void forEachFileInFolder(const std::filesystem::path& folderpath, const std::string& validExtensionList = std::string{}, const std::function<void(const std::filesystem::path&)>& callback = [](const std::filesystem::path& p) { (void)p; }, bool recursive = false) const;
+
+    bool readBufferFromFile(std::vector<unsigned char>& out_buffer, const std::string& filePath) const;
+    bool readBufferFromFile(std::string& out_buffer, const std::string& filePath) const;
+
+    bool writeBufferToFile(void* buffer, std::size_t size, const std::string& filePath) const;
+    bool writeBufferToFile(const std::string& buffer, const std::string& filePath) const;
+
+
+    mutable std::vector<std::filesystem::path> mSearchPath{};
+    std::filesystem::path              mDataPath{};            /**< Data path string. This will typically be 'data/'. */
+    std::string                        mOrganizationName{};    /**< The organization name. Only used for compatibility while transitioning from PhysFS */
+    std::string                        mAppName{};             /**< The application name. Only used for compatibility while transitioning from PhysFS */
+    std::filesystem::path              mWorkingDirectory{};  /**< The working directory. Typically 'mDataPath/mOrganizationName/mAppName/' */
+    std::filesystem::path              mExePath{};           /**< Path to the executable.*/
+    mutable bool                       mVerbose{ false };      /**< Displays lots of messages when true. Otherwise only critical messages are displayed. */
+    mutable bool                       mIsInit{ false };       /**< Has the file system been initialized? */
 };
 
+template<typename DirectoryIteratorType>
+void forEachFileInFolders(const std::filesystem::path& preferred_folderpath, const std::vector<std::string>& validExtensions, const std::function<void(const std::filesystem::path&)>& callback) {
```

Comment:
Brace placement doesn't conform to the rest of the code. It's again a minor thing but I would argue that consistency is important regardless of placement choice. I usually go with brace placement on a new line.